src/irmin-pack/unix/dispatcher_intf.ml

```diff
@@ -50,6 +50,20 @@ module type S = sig
   (** [shrink_accessor_exn a ~new_len] is [a] where the length is smaller than
       in [a].*)
 
+  val create_sequential_accessor_seq :
+    t ->
+    min_header_len:int ->
+    max_header_len:int ->
+    read_len:(bytes -> int) ->
+    (int63 * accessor) Seq.t
+  (** [create_sequential_accessor_seq ~min_header_len ~max_header_len ~read_len]
+      returns a sequence of accessors, which simulates iterating sequentially
+      trough the entries of a pack file. [min_header_len] & [max_header_len]
+      represents the minimum & maximum lengths required to read the header of an
+      entry. [read_len] will then be called with a buffer containing the header
+      of the entry and should return the total length of the entry (the length
+      of he header plus the length of the payload)*)
```

Comment:
Could you add some tests that demonstrate the usage of this new api? In addition to testing the code, it would help me think more about this particular api.

Reply:
My bad, it is meant to be later used in #1939, it is part of the process of getting rid of duplicate code in it.
On the matter of tests however, I have never created any for irmin, so I might not be the best person for them

Reply:
I was looking it up and indeed, it does not seem that the dispatcher is yet directly tested. If we don't mind, we should go for what you proposed but keep in mind that tests will be necessary in a close future, maybe via an issue

Reply:
I agree that it would be nice to have tests for lower levels of the stack -- thanks for opening https://github.com/mirage/irmin/issues/2092. I think it is okay to proceed as-is; perhaps we can pick up this testing as a part of our upcoming GC work.